src/file_watcher.rs

```diff
@@ -0,0 +1,1426 @@
+                                break;
+                            }
+                            _ => panic!("Child process terminated unexpectedly"),
+                        }
+                    }
+                }
+                ForkResult::Child => loop {
+                    sleep(Duration::from_secs(10));
+                },
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod poll_directory {
+    use super::*;
+    use crate::setup_fuelup_home;
+    use rusty_fork::rusty_fork_test;
+    use std::{fs::File, io::Write, sync::OnceLock};
+
+    fn setup_test_file() -> PathBuf {
+        static TEST_FILE: OnceLock<PathBuf> = OnceLock::new();
+
+        let filename = TEST_FILE.get_or_init(|| {
+            let filename = format!("{}/{}", telemetry_config().unwrap().fuelup_tmp, "test.telemetry.file");
+
+            let mut file = File::create(Path::new(filename.as_str())).unwrap();
+            file.write_all(b"MjAyNS0wMy0xNlQxNDo0ODoxMC4wODQ4MDMwMDBaICBJTkZPIGFhcmNoNjQtYXBwbGUtZGFyd2luOkRhcndpbjoyMy42LjAgc2ltcGxlOjAuMS4wOmV4YW1wbGVzL3NpbXBsZS5ycyA0YjEyNDYyNy0wNjNiLTQwM2UtODEyZi02OWQxMzRlZjAxMzIgYXV0bzogZ0Z1ZWwh").unwrap();
+
+            PathBuf::from(filename)
+        });
+
+        filename.clone()
+    }
+
+    rusty_fork_test! {
+        #[test]
+        fn find_telemetry_files_failed() {
+            setup_fuelup_home();
+
+            struct FindTelemetryFilesFailed;
+
+            impl PollDirectoryHelpers for FindTelemetryFilesFailed {
+                fn find_telemetry_files(&mut self, _ignore_age: bool) -> Result<Vec<PathBuf>> {
+                    Err(TelemetryError::Mock)
+                }
+            }
+
+            let file_watcher = FileWatcher::new();
+            let result = file_watcher.poll_directory_with_helpers(&mut FindTelemetryFilesFailed);
+
+            assert_eq!(result, Err(TelemetryError::Mock));
+       }
+
+        #[test]
+        fn find_telemetry_files_empty() {
+            setup_fuelup_home();
+
+            struct FindTelemetryFilesEmpty;
+
+            impl PollDirectoryHelpers for FindTelemetryFilesEmpty {
+                fn find_telemetry_files(&mut self, _ignore_age: bool) -> Result<Vec<PathBuf>> {
+                    Ok(vec![])
+                }
+            }
+
+            let file_watcher = FileWatcher::new();
+            let result = file_watcher.poll_directory_with_helpers(&mut FindTelemetryFilesEmpty);
+
+            assert_eq!(result, Ok(true));
+        }
+
+        #[test]
+        fn flock_failed() {
+            setup_fuelup_home();
+
+            struct FlockFailed;
+
+            impl PollDirectoryHelpers for FlockFailed {
+                fn find_telemetry_files(&mut self, _ignore_age: bool) -> Result<Vec<PathBuf>> {
+                    Ok(vec![setup_test_file()])
+                }
+
+                fn flock(&self, file: File, _flags: FlockArg)-> std::result::Result<Flock<File>, (File, nix::errno::Errno)> {
```

Comment:
kinda weird seeing 2 different `Result` types used... Can we just use the `Result` type instead of the `std` one?